org.eclipse.wildwebdeveloper/src/org/eclipse/wildwebdeveloper/debug/AbstractRunHTMLDebugTab.java

```diff
@@ -100,7 +95,30 @@ public void createControl(Composite parent) {
 				updateLaunchConfigurationDialog();
 			}
 		}));
-
+		
+		Button urlRadio = createRadioButton(resComposite, "URL: "); //$NON-NLS-1$
+		urlRadio.setToolTipText(Messages.RunFirefoxDebugTab_URL_Note);
+		urlRadio.setLayoutData(new GridData(SWT.DEFAULT, SWT.DEFAULT, false, false));
+		urlRadio.addSelectionListener(SelectionListener.widgetSelectedAdapter(e -> {
+			this.programPathText.setEnabled(false);
+			this.filePath.setEnabled(false);
+			this.urlText.setEnabled(true);
+			decoration.hide();
+			setDirty(true);
+			updateLaunchConfigurationDialog();
+		}));
+		urlText = new Text(resComposite, SWT.BORDER);
+		GridData urlTextGD = new GridData(SWT.FILL, SWT.DEFAULT, true, false);
+		urlTextGD.horizontalSpan = 2;
+		urlText.setLayoutData(urlTextGD);
+		urlText.addModifyListener(e -> {
+			setDirty(true);
+			updateLaunchConfigurationDialog();
+		});
+		// Select file debugging by default
+		fileRadio.setSelection(true);
```

Comment:
This creates an annoying bug, which is that the File radio button is always checked, even if the URL radio button was checked previously.

Reply:
Radio selection should be updated in the `initializeFrom()` method then.